test/input.element.test.ts

```diff
@@ -0,0 +1,166 @@
+    const iconElement = el.shadowRoot?.querySelector("sl-icon");
+    iconElement?.setAttribute('name','stack');
+    await elementUpdated(el);
+    expect( iconElement?.getAttribute('name')).to.equal("stack");
+  })
+
+  //Name
+  it("updates the name attribute value to 'Hello'", async () => {
+    const el = await fixture(html`<sgds-input></sgds-input>`);
+    el?.setAttribute('name','Hello');
+    await elementUpdated(el);
+    const name = el.shadowRoot?.querySelector('.form-control')
+    expect( name?.getAttribute('name')).to.equal("Hello");
+  })
+
+  //Placeholder
+  it("updates the default placeholder value to 'Hello'", async () => {
+    const el = await fixture(html`<sgds-input></sgds-input>`);
+    el?.setAttribute('placeholder','Hello');
+    await elementUpdated(el);
+    const placeHolder = el.shadowRoot?.querySelector('.form-control')
+    expect( placeHolder?.getAttribute('placeholder')).to.equal("Hello");
+  })
+
+  it('should focus the input when clicking on the label', async () => {
+    const el = await fixture<SgdsInput>(html` <sgds-input label="Name"></sgds-input> `);
+    const label = el.shadowRoot?.querySelector('label')!;
+    const submitHandler = sinon.spy();
+
+    el.addEventListener('sgds-focus', submitHandler);
+    (label as HTMLElement).click();
+    await waitUntil(() => submitHandler.calledOnce);
+
+    expect(submitHandler).to.have.been.calledOnce;
+  });
+
+  it('should be invalid when the pattern does not match', async () => {
+    const el = await fixture<SgdsInput>(html` <sgds-input pattern="^test" value="fail"></sgds-input> `);
+    expect(el.invalid).to.be.true;
+    expect(el.reportValidity()).to.be.false;
+  });
+
+});
+
+describe('when calling HTMLFormElement.reportValidity()', () => {
+  it('should be invalid when the input is empty and form.reportValidity() is called', async () => {
+    const form = await fixture<HTMLFormElement>(html`
+      <form>
+        <sgds-input required value=""></sgds-input>
+        <sgds-button type="submit">Submit</sgds-button>
+      </form>
+    `);
+
+    expect(form.reportValidity()).to.be.false;
+  });
+
+  it('should be valid when the input is empty, reportValidity() is called, and the form has novalidate', async () => {
+    const form = await fixture<HTMLFormElement>(html`
+      <form novalidate>
+        <sgds-input required value=""></sgds-input>
+        <sgds-button type="submit">Submit</sgds-button>
+      </form>
+    `);
+
+    expect(form.reportValidity()).to.be.true;
+  });
+
+  
+  it('fires sgds-input event when value is entered', async () => {
+    const el = await fixture<SgdsInput>(html` <sgds-input></sgds-input> `);
+    const inputHandler = sinon.spy();
+    const label = el.shadowRoot?.querySelector('label')!;
+    (label as HTMLElement).click();
+    el.addEventListener('sgds-input', inputHandler);
+    await sendKeys({ press: 'A' });
+    waitUntil(()=> inputHandler.calledOnce)
+    expect(inputHandler).to.have.been.calledOnce;
+  });
+
```

Comment:
Will be good to add a test for button and form interactions
1) clicking reset button --> what happens to input
2) clicking submit button --> what happens to input 

For form :
3) sgds-change, sgds-blur events
4) keyboard interactions 
5) watching the disabled 

For button:
6) any new features of button added in sgds-button.ts (blur, focus, and the form attributes (im not sure how it works currently)

Reply:
Added button tests.